README.md

```diff
@@ -1,1105 +1,20 @@
-length fields).  The issue is that the string formatting operator makes
-understanding the output difficult in many cases.   For example in the
-following example, it's difficult to read the code and determine what the
-resulting output will look like:
-
-```python
-print "python %s %s %s/vesselinfodir/ %s/ > /tmp/carter.out 2> /tmp/carter.err"%(carter_script, dist_char, prefix,prefix)
-```
-
-A better way to write this:
-
-```python
-print "python "+carter_script+" "+dist_char+" "+prefix+"/vesselinfodir/ "+prefix+"/ > /tmp/carter.out 2> /tmp/carter.err"
-```
-
-
-Use `else` statements for handling errors, not for normal flow in most cases.
-
-For example, if you have a function that is supposed to take a positive integer
-argument, don't do the following:
-
-```python
-if x > 0:
-   ...
-else: # x must be 0
-   ...
-```
-
-It may be the case that x is negative, a different type, etc.   Instead do the
-following:
-
-
-```python
-if x > 0:
-   ...
-elif x == 0: # x must be 0
-   ...
-else:
-   raise InternalError('Expected x to be a positive integer, not "'+str(x)+'"')
-```
-
-
-## Potpourri ##
-
-### Unix not Windows style text files ###
-
-If you develop on Windows make sure you use `dos2unix` before checking in.
-
-
-### Avoid objects ###
-
-
-Objects are a terrific programming tool when used correctly, but make code
-nearly unreadable when used poorly.   Unfortunately, even experienced
-developers have a hard time knowing how to correctly use objects.   About 80%
-of the code one could write is equally good with and without objects.   10% is
-easier with objects and 10% is easier without them.   Secure Systems Lab code
-should use objects only in very rare cases.   The 90% of the code where objects
-can be avoided without significant impact, should be written without objects.
-
-
-### No lambda functions or lisp-esque code ###
-
-Don't use lambda functions.   Don't use `map`, flatten, etc.   It's best
-not to use these functions and to write longer code that produces the
-same result instead because using these functions makes your code very
-dense and difficult to read.
-
-Do not use syntax like:
-
-```python
-must_testy = [must_test(line) for line in changed.split("\n")]
-```
-
-
-### Write test cases ###
-
-Since you're going to test your module anyways before integrating it into the
-rest of the project (right?), it makes sense to do it in a reusable way.   This
-way when you make changes later, you can immediately test to see if anything is
-broken.
-
-Write 90% tests.   The test should make sure that it will catch 90% of the
-potential problems with the function.   Avoid writing 50% tests (that only
-check a common case or two) and avoid writing 99.9% tests because it will
-consume too much of your time.
-
-All "external" functions **must** be tested.   Internal functions (starting
-with `__`) are up to your discretion.
-
-    
-### Never use short circuit evaluation to produce output
```

Comment:
I am a bit unsure about this recommendation and whether we should keep it (it is marked for removal in this PR).

I find the examples very confusing, apart from distracting syntax errors I have to guess what `askokcancel` does, even though its call requires four lines. And the "YES" examples suggest excessive nesting that does not seem necessary (also see #3).

Is the problem here that `askokcancel` produces output? Or that it blocks on stdin? Is short circuit evaluation generally discouraged?

Reply:
Either way, this PR should close #3.